simulation_utils/cram_decontam_func.py

```diff
@@ -0,0 +1,134 @@
+
+from pysam import VariantFile
+from typing import Tuple
+import os
+import logging
+import pickle
+import hailtop.batch as hb
+import hail as hl
+from .generics import *
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("CHARR simulation pipeline")
+logger.setLevel(logging.INFO)
+
+
+def sample_variant_site(genotype, error, ref, alt1, alt2):
+    p_err = np.random.binomial(1, error)
+    if genotype == (1, 1):
+        allele = random.sample(ALLELES, 1) if p_err else alt1
+    elif genotype == (0, 1):
+        allele = random.sample(ALLELES, 1) if p_err else random.sample((alt1, ref), 1)
+    elif genotype == (1, 2):
+        allele = random.sample(ALLELES, 1) if p_err else random.sample((alt1, alt2), 1)
+    elif genotype == (2, 2):
+        allele = random.sample(ALLELES, 1) if p_err else alt2
+    else:
+        raise Exception("Expecting non reference genotype")
+    return allele
+
+
+def get_var_dict_from_gvcf(input_gvcf: str):
+    # gvcf -> nested dictionary: {chromosome : { position : infomration ...}}
+    # os.system(f"ls {input_gvcf}")
+    print(pysam.__version__)
+    var_dict = {}
+    gvcf_in = VariantFile(input_gvcf)  # auto-detect input format
+    for rec in gvcf_in.fetch():
+        if (rec.samples[0]["GT"][0] != 0) | (rec.samples[0]["GT"][1] != 0):
+            if rec.contig not in var_dict.keys():
```

Comment:
This is fine if you'd like (although you don't have to call `.keys()`, rather `in var_dict` is sufficient), but you should check out `from collections import defaultdict`